doc/config.rst

```diff
@@ -82,6 +82,60 @@ basis set for optimisations:
 
 ------------
 
+Temporary configuration
+**********************
+
+It is also possible to change configuration temporarily, by using the context
+manager:
+
+.. code-block:: python
+
+    >>> ade.Config.ORCA.keywords.opt.functional
+    Functional(pbe0)
+    >>> ade.Config.n_cores = 4
+    >>> mol = ade.Molecule(smiles='CCO')
+    >>> with ade.temporary_config():
+    >>>     ade.Config.n_cores = 9
+    >>>     ade.Config.ORCA.keywords.opt.funcitonal = 'B3LYP'
+    >>>     # this calculation will run with 9 cores and B3LYP functional
+    >>>     mol.optimise(method=ade.methods.ORCA())
+    >>> # when context manager returns previous state of Config is restored
+    >>> ade.Config.n_cores
+    4
+    >>> ade.Config.ORCA.keywords.opt.functional
+    Functional(pbe0)
+
+When the context manager exits, the previous state of the configuration is
+restored.
+
+.. warning::
+    Note that the context manager works by saving the state of the Config
+    when it is called and restoring the state when it exits. The way Python
+    handles object references means that any references taken before or inside
+    the context manager will become useless after it exits. Please see the example
+    below for details.
+
+.. code-block:: python
```

Comment:
does this render ok (with a code-block inside a warning)?

Reply:
No, the code-block is outside the warning. (At least that's how it is rendered now in Pycharm)